src/components/Form/FormSection.tsx

```diff
@@ -155,10 +163,10 @@ export function FormSection<TData extends GenericFormValues>({
       {content &&
         (isEditing ? (
           <form className={styles.flexBox} noValidate onSubmit={(event) => event.preventDefault()}>
-            {content({ values, isEditing, isBusy, onChange, errors: formErrors })}
+            {content({ values, isEditing, isBusy, onChangeEvent, onChangeValue, errors: formErrors })}
```

Comment:
you can handle CustomRegisterFields changes with onChange, I am not sure that we need separate methods

Reply:
The `<DateField />` component inside `<CustomRegisterField />` is the reason for this. The `onChange` callback relies on an actual event object - the React-specific `SyntheticEvent` one, but `<DateField onChange />` doesn't produce such event because internally it doesn't represent a single element which can be an event target, which makes it incompatible with onChange (that's why I renamed it to `onChangeEvent`). I had thought about `new Event('change')`, but the native JS `Event` object is still incompatible with `SyntheticEvent`, while React doesn't offer `new SyntheticEvent()`, it's just a typescript type.